micropub.php

```diff
@@ -481,6 +496,37 @@ private static function handle_authorize_error( $code, $msg ) {
 		static::respond( $code, $msg );
 	}
 
+	private static function post_status( $mf2 ) {
+		$props = $mf2['properties'];
+		if ( isset( $props['visibility'] ) ) {
+			if ( array( 'private' ) === $props['visibility'] ) {
+				return 'private';
+			}
+			// Public or Private are the only options
+			if ( array( 'public' ) !== $props['visiblity'] ) {
+				return null;
+			}
+		}
+		if ( ! isset( $props['post-status'] ) ) {
+			return get_option( 'micropub_default_post_status' );
+		} else {
+			//  According to the proposed specification these are the only two properties supported.
+			// https://indieweb.org/Micropub-extensions#Post_Status
+			// For now these are the only two we will support even though WordPress defaults to 8 and allows custom
+			// But makes it easy to change
+			$statuslist = array( array( 'published' ), array( 'draft' ) );
+			if ( ! in_array( $props['post-status'], $statuslist, true ) ) {
+				return null;
+			}
+			// Map published to the WordPress property publish.
+			if ( 'published' === $props['post-status'] ) {
+				return 'publish';
+			}
+			return $props['post-status'];
```

Comment:
this returns an array, right? probably should be just `return 'draft';`? sounds like we need another unit test. :P

Reply:
I have one for the creation of draft status. It seems to work. What would you suggest the other unit test does?

Reply:
ah, that's true, thanks! that's a good enough test. i still don't understand this code, though. `$props['post-status']` is always an array here, right? so the conditional on line 522 will always be false, and this line will always return that array...so then when we set `$args['post_status']` to an array instead of a string on line 364, and then call `wp_update_post()`, wordpress handles the array value ok?